platform/search/src/services/BareBonesSearchWorker.js

```diff
@@ -43,28 +44,46 @@
      *   input. Returns matching results from indexedItems
      *
      * @param data An object which contains:
-     *           * input: The original string which we are searching with
+     *           * input: The original string which we are searching with,
+     *           * or an object with 2 properties, searchString and typeFilter
      *           * maxResults: The maximum number of search results desired
      *           * queryId: an id identifying this query, will be returned.
      */
     function search(data) {
+        let searchString,
+            typeFilter;
+
+        if (typeof data.input === 'object') {
+            searchString = data.input.searchString;
+            typeFilter = data.input.typeFilter;
+        } else if (typeof data.input === 'string') {
+            searchString = data.input;
+            typeFilter = '';
+        }
+
+
         // This results dictionary will have domain object ID keys which
         // point to the value the domain object's score.
-        var results,
-            input = data.input.trim().toLowerCase(),
+        let filteredByType,
+            filteredByName,
+            trimmedSearchString = searchString.trim().toLowerCase(),
             message = {
                 request: 'search',
                 results: {},
                 total: 0,
                 queryId: data.queryId
             };
 
-        results = indexedItems.filter((indexedItem) => {
-            return indexedItem.name.includes(input);
+        filteredByType = indexedItems.filter((indexedItem) => {
+            return indexedItem.type.includes(typeFilter);
+        });
+
+        filteredByName = filteredByType.filter((indexedItem) => {
+            return indexedItem.name.includes(trimmedSearchString);
```

Comment:
Given you always filter by name and type, you can retain results and just combine the two filter calls.

```suggestion
        results = indexedItems.filter(item => item.type.includes(typeFilter) && item.name.includes(trimmedSearchString));
```

Reply:
Ah good catch, this was all done in 5 mins, so its a little rushed, ill change that